cmd/hub/mcscontroller/controller.go

```diff
@@ -117,16 +124,138 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 // - ServiceImports are owned by the corresponding ServiceExport and are thus
 //	 automatically deleted by k8s when their ServiceExport is deleted.
 //
-func (r *ServiceExportReconciler) reconcile(ctx context.Context, name types.NamespacedName) (ctrl.Result, error) {
-	_ = r.Log.WithValues("service", name)
+func (r *ServiceExportReconciler) reconcile(ctx context.Context, name types.NamespacedName, listOptions *client.ListOptions) (ctrl.Result, error) {
+	log := r.Log.WithValues("service", name)
+	var exports mcsv1a1.ServiceExportList
+	if err := r.Client.List(ctx, &exports, listOptions); err != nil {
+		log.Error(err, "unable to list service's service export")
+		return ctrl.Result{}, err
+	}
+	primary, err := getPrimaryExportObject(exports)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	for _, se := range exports.Items {
+		if se.DeletionTimestamp != nil {
+			continue
+		}
 
-	//------------------------------------------
-	// -- @TODO CODE ABOVE SPECIFIED BEHAVIOR --
-	//------------------------------------------
+		exportSpec := &mcs.ExportSpec{}
+		err := exportSpec.UnmarshalObjectMeta(&se.ObjectMeta)
+		if err != nil {
+			log.Info("Failed to unmarshal exportSpec from serviceExport")
+			return ctrl.Result{}, err
+		}
 
+		err = exportSpec.IsCompatibleWith(primary)
+		if err != nil {
+			lhutil.UpdateServiceExportConditions(ctx, &se, log, mcsv1a1.ServiceExportConflict, corev1.ConditionTrue, "", err.Error())
+			r.Client.Status().Update(ctx, &se)
+			si := mcsv1a1.ServiceImport{}
+			err = r.Client.Get(ctx, name, &si)
```

Comment:
shouldnt you only delete the import for this cluster?

Reply:
@davidohana when you detect a change in the export you should clear all (now) incompatible imports, even if their export is not changed.